internal/pkg/policy/policy_output.go

```diff
@@ -229,6 +233,35 @@ func (p *Output) prepareElasticsearch(
 		output.PermissionsHash = p.Role.Sha2 // for the sake of consistency
 	}
 
+	if outputMap.GetMap(p.Name).GetString("type") == OutputTypeRemoteElasticsearch {
+		om, _ := outputMap.GetMap(p.Name).Marshal()
+		var outputObj map[string]any
+		json.Unmarshal(om, &outputObj)
+		hosts := outputObj["hosts"].([]interface{})
+		outputAPIKey, err :=
+			generateRemoteOutputAPIKey(ctx, agent.Id, p, []string{hosts[0].(string)})
```

Comment:
It was pretty cumbersome to get `hosts` out of the `outputMap`, I appreciate any suggestions to do it simpler.

Reply:
I think you're in the right path when you defined the json tag. You might need to define different models for each output type.

Also there is another thing to consider here, ideally I believe it'd be good to try all the hosts if they fail to generate the API key